### ISSUE TITLE
Adding automation to flathub/flatpak builds

### ISSUE THREAD

**Issue by tristanvigar** [organicmaps/organicmaps#8337]
Opening on recommendation from #8331 

Is there a mechanism within Github or elsewhere that could be leveraged to automate .metainfo.xml updates? This would remove a step required for flatpak releases. 

If somebody could make a recommendation, I could look at introducing this functionality.

Thanks!

cc: @Ferenc- as I've noticed they tend to handle flatpak releases.

**Comment by Ferenc-** [comment]
> Is there a mechanism within Github or elsewhere that could be leveraged to automate .metainfo.xml updates?

The current versioning scheme is such that the the release version and the tag itself is dependent on the number of git commits on that given day. Consequently the `.metainfo.xml`, can only be updated with 100% certainty after the release. Which means that  in the flatpak repo you would also have to automatically create a new patch, with the retroactive `.metainfo.xml` changes. 
So currently it wouldn't be enough to just automate the .metainfo.xml update.

An alternative would be to somehow change to a scheme where the `.metainfo.xml` could be changed before the release,
that would require an approval from the project leaders.

**Comment by biodranik** [comment]
What kind of scheme change is needed?

**Comment by Ferenc-** [comment]
> What kind of scheme change is needed?

The way I see:
1. Tag a release with the release changelog already in the `.metainfo.xml`.
2. In order to make `1.` possible, don't use the number of commits in the release tag.

**Comment by biodranik** [comment]
A separate tag for desktop builds?

**Comment by Ferenc-** [comment]
> A separate tag for desktop builds?

Yes, that would be a potential solution, provided that tag is on a commit which already has the `.metainfo.xml` change.
